### PR TITLE
Add custom timeout option - Issue #20

### PR DESCRIPTION
**Pull request by jthampton**
I have done integration testing locally, and this seems to be working as I hoped.  I don't know how to create an actual unit test considering this is affecting the actual connection settings.  I did update a failing test in the cisco base around the prompt, and I did add examples on how to use the custom timeout.  I have also updated the README with an example as well.

Please let me know if you have any questions/concerns.

### CODE REVIEW

**Comment by Ali-aqrabawi** [comment]
can you please use `DeviceOption` to set timeout, 

you should be able to call it like follows:

```go
device, err := gomiko.NewDevice("192.168.1.1",
                                "admin",  
                                "password", 
                                "cisco_ios", 22,
                                gomiko.SecretOption("enablePass"), 
                                gomiko.TimeOutOption(10))
```
or like below :
```go
device, err := gomiko.NewDevice("192.168.1.1",
                                "admin",  
                                "password", 
                                "cisco_ios", 22,
                                gomiko.TimeOutOption(10))
```
refer to `package/options.go`

**Comment by jthampton** [comment]
I think this would require a refactor considering the timeout option is defined as part of creating the connection.  The connection has to be defined before you create the device and the options are defined for the newly created device (`pkg/deviceFactory.go`).  

I understand that the TimeOutOption would make sense as being part of the optional parameter Options to NewDevice, however, I'm not sure how to pull that off to be part of the NewConnection call, where the timeout needs to go.

Thanks for the suggestion, however, I'm not sure how to proceed.

**Comment by Ali-aqrabawi** [comment]
>>The connection has to be defined before you create the device and the options are defined for the newly created device `(pkg/deviceFactory.go)`.

yes the required paramters are initialized at `deviceFactory.go` , and the optionals we add them with setters methods.

i hope this give some insight how we can achieve this,

`types.Device` should have `setTimeOut` method like :

```go
type Device interface {
	Connect() error
	Disconnect()
	SendCommand(cmd string) (string, error)
	SendConfigSet(cmds []string) (string, error)
        SetTimeout(secret int)
}
```
`base.device` which is used as abstract type for all devices:

```go
func (d *CSCODevice) SetTimeOut(timeout int) {
	d.Driver.Connection.setTimeOut(timeout)
}
```

`ssh.go`:

```go
type SSHConn struct {
	addr     string
	username string
	password string
	client   *ssh.Client
	reader   io.Reader
	writer   io.WriteCloser
        timeout int
}

func (c *SSHConn) SetTimeout(timeout int) {

	c.timeout = timeout

}

```

and `options.go`:

```go
func TimeOutOption(timeout int) func(device interface{}) error {
	return func(device interface{}) error {
		device.(types.CiscoDevice).setTimeOut(timeout)
		return nil
	}
}
```

**Comment by jthampton** [comment]
I appreciate the guidance.  I've updated the code and all the tests so that they run properly with the new `TimeoutOption`.  I've run an integration test against a Cisco ASA device and it works great.  I've debugged the connection to ensure the timeout updates if using the TimeoutOption or remains set at 6 by default.

Let me know if you have other concerns or suggestions.

File: README.md [hunk 1]
```diff
@@ -83,7 +83,9 @@ import (
 
 func main() {
 	
-     device, err := gomiko.NewDeviceWithTimeout("192.168.1.1", "admin", "password", "cisco_ios", 22, 10)
+     secret := gomiko.SecretOption("enablePass")
+     timeout := gomiko.TimeoutOption(10)  // Default is 6
+     device, err := gomiko.NewDevice("192.168.1.1", "admin", "password", "cisco_ios", 22, secret, timeout)
```
Comment (Ali-aqrabawi) [inline README.md]: please use this instead:

```go
device, err := gomiko.NewDevice("192.168.1.1", "admin", "password", "cisco_ios", 22, 
                                 gomiko.SecretOption("enablePass"), 
                                 gomiko.TimeoutOption(10))
```

File: pkg/connections/ssh.go [hunk 2]
```diff
@@ -26,26 +26,26 @@ type SSHConn struct {
 	client   *ssh.Client
 	reader   io.Reader
 	writer   io.WriteCloser
-	timeout  uint8
```
Comment (Ali-aqrabawi) [inline pkg/connections/ssh.go]: `uint8`  is more suitable here, i used `int` in my example for the sake of the example. sorry for confusion.

**Review by Ali-aqrabawi — COMMENTED** [review]
i have added two minor comments. please take a look.

**Comment by jthampton** [comment]
I've updated the type for `timeout` from `int` to `uint8` and updated the readme to keep the examples consistent.

**Pull request merged**